vars/checkPatch.groovy

```diff
@@ -51,8 +52,14 @@ def call(Map config = [:]) {
         ignored_files += ":" + config['ignored_files']
     }
 
+    def checkpatch_args="--notree --show-types -"
+    if (config['codespell_file']) {
+       checkpatch_args="--notree --show-types --codespell --codespellfile=" + config['codespell_file'] + " -"
+    }
+
     int rc = 1
     def script = 'CHECKPATCH_IGNORED_FILES="' + ignored_files + '"' + \
+                 ' CHECKPATCH_ARGS="' + checkpatch_args + '"' + \
```

Comment:
This does work, but the script this function calls is jenkins_github_checkwarn.sh which I seems to build a default CHECKPATCH_ARGS if one isn't set, and this is where the ignored_files option is used, so I think this PR probably disables that option.

An alternative would be to modify github_checkpatch.py to insert the codespell options there, based on availability of the package.

Reply:
I wanted the checkpatch stage and scripts to be configured to use optional files to control what is checked and what is ignored, sort of like the pylint rc files, rather than passing a list to it.

I have not looked at the current check patch code in a few years now so I need to look up the impact of this change.

Reply:
Yes, it's far from clean the way checkpatch currently works.  Trying a different approach elsewhere but I do appreciate advise on the best way forward here.

Reply:
`code_review/jenkins_github_checkwarn.sh` already has _codespell_ support in it?

Reply:
checkpatch.pl already has support in it, it just needs the name of the file on the command line.